src/service/chat.js

```diff
@@ -245,19 +254,18 @@ const getAllowAnonDmStatus = async (postId) => {
   }
 };
 
-const initChatFromPostAnon = async ({source, id}) => {
+const initChatFromPostAnon = async ({source, id, feedId}) => {
   let payload = {source};
   switch (source) {
     case 'post':
       payload = {...payload, postId: id};
       break;
     case 'comment':
-      payload = {...payload, commentId: id};
+      payload = {...payload, commentId: id, postId: feedId};
       break;
     default:
       break;
   }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This change mirrors the one made in the `initChatFromPost` function, and the same comments apply. The addition of `feedId` as `postId` in the 'comment' case seems logical. Again, consider adding error handling for the case when `source` is neither 'post' nor 'comment'.

```diff
+    default:
+      throw new Error(`Invalid source: ${source}`);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->